### ISSUE TITLE
Large Rec File Memory and Speed Issues

### ISSUE THREAD

**Issue by RishabGargeya** [apache/incubator-mxnet#3183]
Hi all,

I just had a question about the usage of recordio files in mxnet. I have a large .rec file (~35 GB) on a system with 32GB of RAM. Whenever I try training a model, I see my memory consumption overflowing into swap memory, eating around as much RAM as the .rec file itself. Changing prefetch_buffer and preprocess_threads constants don't seem to change this. Is mxnet loading the entire rec file into memory? I saw this issue https://github.com/dmlc/mxnet/issues/1411 which is similar to my issue, but working off of mxnet's latest commit didn't solve.

Also curiously, I am still able to train albeit slowly, but running the validation set between epochs takes >3 hrs when each epoch is only about 1 hr. Is this normal?

Thanks so much for the help!

**Comment by tornadomeet** [comment]
hi, i trained 103G .rec file, and the RAM using is less than 20G.
if using one GPU, for your big dataset, it takes > 3 hrs is normal.

**Comment by KeyKy** [comment]
I have the same problem when i train ssd or imagenet and i check the usage of memory in top. At the beginning of training is 5.7% , after a while (3~4h) it takes 27% and finally it run out of my memory (waked up by the alarm message)

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!
Also, do please check out our [forum](https://discuss.mxnet.io/) (and [Chinese version](https://discuss.gluon.ai/)) for general "how-to" questions.
